src/WebJobs.Extensions.DurableTask/Listener/DurableTaskScaleMonitor.cs

```diff
@@ -39,7 +42,15 @@ internal sealed class DurableTaskScaleMonitor : IScaleMonitor<DurableTaskTrigger
             this.storageAccount = storageAccount;
             this.logger = logger;
             this.performanceMonitor = performanceMonitor;
+            this.durableTaskMetricsProvider = durableTaskMetricsProvider;
+
+#if FUNCTIONS_V3_OR_GREATER
+            this.scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{this.functionId}-DurableTaskTrigger-{this.hubName}".ToLower(), this.functionId);
+#else
+#pragma warning disable CS0618 // Type or member is obsolete

```

Comment:
Can you improve the comment to explain why it's okay for us to disable the obsolete warning?

Reply:
I added a comment. Essentially, this new constructor is not compatible with Functions V1 -> V3. Unfortunately, we don't have a way to exclude that whole group, but we can exclude Functions V1 and V2 and have those use the old, obsolote, constructor. That's what we're doing here.